File: day1/c++/day1.cpp
#include <iostream>
#include "./day1_lib.h"
#include <fstream>

int main() {
    std::cout << "----- Part 1 -----" << std::endl;

    std::ifstream input_file("../../input/1");

    int total_fuel = 0;
    int mass = 0;

    while(input_file >> mass) {
        total_fuel += calc_fuel(mass);
    }

    std::cout << "Total fuel is " << total_fuel << std::endl;

    std::cout << "----- Part 2 -----" << std::endl;

    // Return to start of file
    input_file.clear();
    input_file.seekg(0, std::ios::beg);
    total_fuel = 0;

    while(input_file >> mass) {
        total_fuel += recursive_fuel(mass);
    }

    std::cout << "Total fuel is " << total_fuel << std::endl;
}

File: day01/ruby/day1.rb
require "./day1_lib.rb"

puts "----- Part 1 -----"

@total_fuel = 0

File.foreach('../../input/1') do |line|
  @total_fuel += calc_fuel(line.to_f)
end

puts "Total fuel needed is: #{@total_fuel}"

puts "----- Part 2 -----"

@total_fuel = 0

File.foreach('../../input/1') do |line|
  @total_fuel += recursive_fuel(line.to_f)
end

puts "Total fuel needed is: #{@total_fuel}"
